### DISCUSSION TITLE
`enableMaximizedFloatingWidget`/`enableMaximizedPanelWidget`: Give us your thoughts!

### DISCUSSION THREAD

**Discussion by raplemie** [iTwin/appui#673] [Preview Features]
# Feature description

When `enableMaximizedFloatingWidget` is enabled the floating widgets[^1] will now have a "maximize" button that allow it to take the complete space used by AppUI, once maximized, a "restore" button will take its place so the widget goes back to it's original place. Note that all tabs from the container are available and are still operational, tabs can be swapped while maximized.

The basis for this request was to enable easy viewing of more data dense widgets with a single click.

When `enableMaximizedPanelWidget` is enabled, the panel widgets will now have a "maximize" button as well.

[^1]: Note that the goal would be that the maximize button would be available on stage panel sections as well when they are docked and have the same effect, but for preview features simplicity, this was only initially implemented on floating widget.
**EDIT:** `enableMaximizedPanelWidget` preview feature is added as well, which when enabled adds the same capability to panel widgets.

[Live demo](https://itwin.github.io/appui/storybook/?path=/story/previewfeatures-enablemaximizedwidget--default)

> [!NOTE]  
> `enableMaximizedFloatingWidget` initially released in 4.7.0
> `enableMaximizedPanelWidget` initially released in 4.10.0

![maximize](https://github.com/iTwin/appui/assets/849817/23378467-2dfa-45d9-b18a-b0a964510027)

## Why is it a preview?

- The top right corner of the widget already contain buttons that affect either only the active tab (popout) or the container itself (send back), so the placement of this button needs to be validated.
- When maximized, the widget completely hides everything below it, rendering the behavior questionable, are there real use-case where a widget in "full screen" is useful ?

## What then?

We would like the community to chime in on any issues they are facing when using this feature, and discuss how we can resolve these. We'd also like to know if there is a particular API that would be useful with this feature, additional options to be considered, modified or additional behavior or any other ideas or opinions on the subject.

Basically, feel free to add comment on usefulness and how this feature should be officially supported!
